netsocadmin/mysql.py

```diff
@@ -137,6 +137,40 @@ def create_user(username: str) -> str:
         con.close()
 
 
+def update_password(username: str) -> str:
+    """
+    update_password changes a user's in the MySQL DBMS to a new random password
+    if and only if a user of that name does already exists.
+
+    :param username the requested username whose password to update.
+    :raises UserError if the operation fails.
+    :returns string the generated password for the new user
+    """
+    if not re.match(config.VALID_USERNAME, username):
+        raise BadUsernameError(f"invalid username '{username}', must be alphanumeric, underscores and hyphens only")
+    try:
+        con = _mysql_connection()
+        with con.cursor() as cur:
+            # check is username already exists
+            sql = "SELECT User FROM mysql.user WHERE User=%s;"
+            cur.execute(sql, username)
+            if not cur.rowcount:
+                raise Exception(f"username {username} doesn't exist")
+
+            # create new password
+            chars = string.ascii_letters + string.digits
+            password = "".join(random.choice(chars) for _ in range(random.randint(10, 15)))
+            sql = """ALTER USER %s@'%%' IDENTIFIED BY %s;"""
+            cur.execute(sql, (username, password,))
+            if not config.FLASK_CONFIG["debug"]:
+                sql = """ALTER USER %s@'localhost' IDENTIFIED BY %s;"""
+                cur.execute(sql, (username, password,))
+            return password
+    except Exception as e:
+        raise UserError(f"failed to create the new user {username}") from e
+    finally:
+        con.close()
```

Comment:
See if you can use transactions for this. Should be an example of it somewhere in the codebase :)